mdm-frontend/src/app/legacy/common/i18n/directives/tageditor/tag-editor-elsst.controller.js

```diff
@@ -0,0 +1,119 @@
+/* global _ */
+'use strict';
+
+angular.module('metadatamanagementApp')
+  .controller('TagEditorElsstController', [
+  '$scope',
+  '$mdMedia',
+  'DataPackageSearchService', 'ElsstSearchService', function($scope, $mdMedia, DataPackageSearchService, ElsstSearchService) {
+
+    var removeExistingTags = function(tags, language) {
+    // return all selectable tags that are available according to the search text and have not already selected
+      if (!$scope.tags[language] || $scope.tags[language].length == 0) {
+        $scope.tags = {
+          de: [],
+          en: []
+        };
+        return tags;
+      }
+
+      return tags.filter(t => !$scope.tags[language].some(i => t.localname === i.localname));
+    };
+
+    var getOppositeLanguage = function(language) {
+      var oppositeLanuage;
+      if (language === 'en') {
+        oppositeLanuage = 'de';
+      } else if (language === 'de') {
+        oppositeLanuage = 'en';
+      } else {
+        console.error('Unknown language', language);
+        return;
+      }
+
+      return oppositeLanuage;
+    }
+
+    if (angular.isUndefined($scope.readonly)) {
+      $scope.readonly = false;
+    }
+    
+    if (angular.isUndefined($scope.tags)) {
+      $scope.tags = {
+        de: [],
+        en: []
+      };
+    } else if (angular.isUndefined($scope.tags.de)) {
+      $scope.tags.de = [];
+    } else if (angular.isUndefined($scope.tags.en)) {
+      $scope.tags.en = [];
+    }
+
+    $scope.selectedItemChange = function(item, language) {
+      // translate a german selected item to english and add it to the english selected tags, and vice verca
+      var translateLang = getOppositeLanguage(language);
+      if (!item || !translateLang) {
+        return;
+      }
+
+      ElsstSearchService.findTagsElsstTranslation(item.prefLabel, language, translateLang)
+        .then(function(response) {
+          if (!response) {
+            return;
+          }
+          for (var i = 0; i < response.length; i++) {
+            // in the other language, sometimes multiple tags are returned, add all of them   // TODO REMOVE SELBE ID ??
+            res = response[i];
+            if (res.altLabel && typeof res.altLabel == 'string') {
+              // sometimes altLabel is originally not given as array, but as string
+              res.altLabel = [altLabel];
+            }
+            if ($scope.tags[translateLang].length == 0) {
+              $scope.tags[translateLang] = [res];
+            } else {
+              $scope.tags[translateLang] = [...$scope.tags[translateLang], res];
+            }
+          }
+        })
+        .catch(function(error) {
+          console.error(error);
+        });
+    };
+
+    $scope.removeInvalidInput = function(chip, language) {
```

Comment:
If you enter a valid tag and confirm with the ENTER key, this tag is removed by the `removeInvalidInput` function. Is this behavior intended? Should you really only be able to select a tag with a mouse click?
![enter-removeInvalidInput](https://github.com/dzhw/metadatamanagement/assets/118182853/0ae3d78f-6cd4-4f7a-952d-6fd9e6160c76)